src/new/wallet.js

```diff
@@ -1008,30 +1140,165 @@ class HathorWallet extends EventEmitter {
+   *
+   * @return {Object} Object with {success: true, sendTransaction, promise}, where sendTransaction is a
+   * SendTransaction object that emit events while the tx is being sent and promise resolves when the sending is done
+   **/
+  delegateAuthority(tokenUid, type, destinationAddress, options = {}) {
+    storage.setStore(this.store);
+    const newOptions = Object.assign({ createAnother: true, startMiningTx: true }, options);
+    const { createAnother, startMiningTx } = newOptions;
+    let filterUtxos;
+    if (type === 'mint') {
+      filterUtxos = wallet.isMintOutput.bind(wallet);
+    } else if (type === 'melt') {
+      filterUtxos = wallet.isMeltOutput.bind(wallet);
+    } else {
+      throw new Error('This should never happen.')
+    }
+
+    const delegateInput = this.selectAuthorityUtxo(tokenUid, filterUtxos);
+
+    if (delegateInput.length === 0) {
+      return {success: false, message: 'Don\'t have authority output available.'}
+    }
+
+    const { tx_id, index, address } = delegateInput[0];
+
+    const ret = tokens.delegateAuthority(tx_id, index, address, tokenUid, destinationAddress, createAnother, type, this.pinCode);
+
+    if (ret.success) {
+      const sendTransaction = ret.sendTransaction;
+      if (startMiningTx) {
+        sendTransaction.start();
+      }
+      return {success: true, promise: sendTransaction.promise, sendTransaction};
+    } else {
+      return ret;
+    }
+  }
+
+  /**
+   * Destroy authority
+   *
+   * @param {String} tokenUid UID of the token to delegate the authority
+   * @param {String} type Type of the authority to delegate 'mint' or 'melt'
+   * @param {number} count How many authority outputs to destroy
+   * @param {Object} options Options parameters
+   *  {
+   *   'startMiningTx': boolean to trigger start mining (default true)
+   *  }
+   *
+   * @return {Object} Object with {success: true, sendTransaction, promise}, where sendTransaction is a
+   * SendTransaction object that emit events while the tx is being sent and promise resolves when the sending is done
+   **/
+  destroyAuthority(tokenUid, type, count, options = { startMiningTx: true }) {
+    storage.setStore(this.store);
+    let filterUtxos;
+    if (type === 'mint') {
+      filterUtxos = wallet.isMintOutput.bind(wallet);
+    } else if (type === 'melt') {
+      filterUtxos = wallet.isMeltOutput.bind(wallet);
+    } else {
+      throw new Error('This should never happen.')
+    }
+
+    const destroyInputs = this.selectAuthorityUtxo(tokenUid, filterUtxos, { many: true });
+
+    if (destroyInputs.length < count) {
+      return {success: false, message: `Don't have enough authority output available to destroy. Have ${destroyInputs.length} and requested ${count}.`}
+    }
+
+    const data = [];
+    for (const utxo of destroyInputs) {
+      const { tx_id, address, index } = utxo;
+      data.push({ tx_id, address, index, token: tokenUid });
+      // Even though count is expected as a number, I am using ==
```

Comment:
I think it is best to use `>=` and avoid infinite loops. We can do a type check at the beginning of the method to assert it has the right type.